Add Testcontainers for .NET documentation (#627)
testcontainers/testcontainers-dotnet: docs/custom_configuration/index.md

# Custom Configuration

> Testcontainers for .NET supports various configurations to set up your test environment. It automatically discovers the Docker environment and applies the configuration. Set or override the default values either with the Testcontainers [properties file][properties-file-format] (`~/testcontainers.properties`) or with environment variables. If you prefer to configure your test environment at runtime, set or override the configuration through the `TestcontainersSettings` class. The following configurations are available:

Reviewer: ```suggestion
Testcontainers supports various configurations to set up your test environment. It automatically discovers the Docker environment and applies the configuration. You can set or override the default values either with the Testcontainers [properties file][properties-file-format] (`~/testcontainers.properties`) or with environment variables. If you prefer to configure your test environment at runtime, you can set or override the configuration through the `TestcontainersSettings` class. The following configuration parameters are available:
```

> | Properties File         | Environment Variable                   | Description                                                                                                               |

Reviewer: We should add the default values to the table (if they make sense).

Author: I've added them.

| `docker.auth.config`    | `DOCKER_AUTH_CONFIG`                   | The Docker configuration file content (GitLab: [Use statically-defined credentials][use-statically-defined-credentials]). |
| `docker.cert.path`      | `DOCKER_CERT_PATH`                     | The directory path that contains the client certificate (`{ca,cert,key}.pem`) files.                                      |
| `docker.tls`            | `DOCKER_TLS`                           | Enables TLS.                                                                                                              |
| `docker.tls.verify`     | `DOCKER_TLS_VERIFY`                    | Enables TLS verify.                                                                                                       |
| `ryuk.disabled`         | `TESTCONTAINERS_RYUK_DISABLED`         | Disables Ryuk (resource reaper).                                                                                          |
| `ryuk.container.image`  | `TESTCONTAINERS_RYUK_CONTAINER_IMAGE`  | The Ryuk (resource reaper) Docker image.                                                                                  |

> | `hub.image.name.prefix` | `TESTCONTAINERS_HUB_IMAGE_NAME_PREFIX` | The name to substitute the Docker Hub image name.                                                                         |

Reviewer: ```suggestion
| `hub.image.name.prefix` | `TESTCONTAINERS_HUB_IMAGE_NAME_PREFIX` | The name to use for substitunging the Docker Hub registry part of the image name.                                                                         |
```

## Enable logging

> To enable logging, set the `TestcontainersSettings.Logger` property to an instance of an `ILogger` implementation. Either use your implementation or choose one of the common third-party providers.

Reviewer: ```suggestion
To enable logging, set the `TestcontainersSettings.Logger` property to an instance of an `ILogger` implementation. Either use your own implementation or choose one of the common third-party providers.
```

### xUnit.net

> The xUnit.net decorates are a convenient way to initialize your test environment just before the first test run. This includes logging. Configure your third-party provider as for any other .NET application, e.g. through the file or environment variables configuration provider. Set up and tear down your test environment with help of `Microsoft.Extensions.Hosting` that encapsulates resources and lifetime functionalities:

Reviewer: ```suggestion
The xUnit.net decorators are a convenient way to initialize your test environment just before the first test run. This includes logging. Configure your third-party provider as for any other .NET application, e.g. through the file or environment variables configuration provider. Set up and tear down your test environment with help of `Microsoft.Extensions.Hosting` that encapsulates resources and lifetime functionalities:
```
